lib/Common/Memory/WriteBarrierMacros.h

```diff
@@ -17,6 +17,8 @@
 
 #define FieldWithBarrier(type, ...) \
     typename WriteBarrierFieldTypeTraits<type, ##__VA_ARGS__>::Type
+#define FieldWithBarrierT(type, ...) \
```

Comment:
@leirocks Thanks, that reduces the changes required. I have changed the macro to not include `typename`. I have added `typename` explicitly for declarations with template dependent name. This is required by both legacy and non legacy.
Submitted a new commit consisting of the changes.